sdk/Lusid.Sdk.Tests/tutorials/Ibor/PortfolioCashFlows.cs

```diff
@@ -0,0 +1,128 @@
+
+        [OneTimeSetUp]
+        public void SetUp()
+        {
+            _apiFactory = LusidApiFactoryBuilder.Build("secrets.json"); ;
+            _transactionPortfoliosApi = _apiFactory.Api<ITransactionPortfoliosApi>();
+            _testDataUtilities = new TestDataUtilities(
+                _apiFactory.Api<ITransactionPortfoliosApi>(),
+                _apiFactory.Api<IInstrumentsApi>(),
+                _apiFactory.Api<IQuotesApi>(),
+                _apiFactory.Api<IStructuredMarketDataApi>());
+        }
+
+        [TestCase(true)]
+        [TestCase(false)]
+        public void ExamplePortfolioCashFlowsForFxForwards(bool isNdf)
+        {
+            // CREATE portfolio
+            var portfolioScope = Guid.NewGuid().ToString();
+            var portfolioId = _testDataUtilities.CreateTransactionPortfolio(portfolioScope);
+
+            // CREATE Fx Forward
+            var fxForward = InstrumentExamples.CreateExampleFxForward(isNdf) as FxForward;
+        
+            // UPSERT Fx Forward to portfolio and populating stores with required market data
+            var effectiveAt = new DateTimeOffset(2020, 2, 23, 0, 0, 0, TimeSpan.Zero);
+            _testDataUtilities.AddInstrumentsTransactionPortfolioAndPopulateRequiredMarketData(
+                portfolioScope, 
+                portfolioId,
+                effectiveAt,
+                effectiveAt,
+                fxForward);
+            
+            // CALL api to get cashflows at maturity
+            var maturity = fxForward.MaturityDate.Value;
+            var cashFlowsAtMaturity = _transactionPortfoliosApi.GetPortfolioCashFlows(
+                portfolioScope,
+                portfolioId,
+                maturity.AddMilliseconds(-1),
+                maturity.AddMilliseconds(1));
+            
+            // CHECK correct number of cashflow at maturity
+            var expectedNumber = isNdf ? 1 : 2;
+            Assert.That(cashFlowsAtMaturity.Values.Count, Is.EqualTo(expectedNumber));
+            
+            var cashFlows = cashFlowsAtMaturity.Values.Select(cf => cf)
+                .Select(cf => (cf.PaymentDate, cf.Amount, cf.Currency))
+                .ToList();
+
+            var expectedCashFlows = isNdf
+                ? new List<(DateTimeOffset? PaymentDate, decimal? Amount, string Currency)>
+                {
+                    (fxForward.MaturityDate, fxForward.DomAmount, fxForward.DomCcy)
+                }
+                : new List<(DateTimeOffset? PaymentDate, decimal? Amount, string Currency)> 
+                {
+                    (fxForward.MaturityDate, fxForward.DomAmount, fxForward.DomCcy),
+                    (fxForward.MaturityDate, fxForward.FgnAmount, fxForward.FgnCcy),
+                };
+
+            Assert.That(cashFlows, Is.EquivalentTo(expectedCashFlows)); 
+        }
+        
+        [TestCase(true)]
+        [TestCase(false)]
+        public void ExampleUpsertablePortfolioCashFlowsForFxForwards(bool isNdf)
```

Comment:
for this example we should show how to get the cashflows out and push them back into the portfolio given that (I believe) this is what this call is for